src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -496,6 +497,15 @@ public override Type[] GetOptionalCustomModifiers()
                 m_signature.GetCustomModifiers(PositionImpl + 1, false);
         }
 
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
```

Comment:
Same as change to `ReferenceEquals` as https://github.com/dotnet/runtime/pull/73009/files#r935802561